src/Http/Controllers/VatController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Rapidez\VatValidation\Http\Controllers;
+
+use Ibericode\Vat\Validator;
+use Ibericode\Vat\Vies\ViesException;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+
+class VatController
+{
+    /** @return array<string, mixed> */
+    public function __invoke(Request $request): array
+    {
+        $request->validate([
+            'id' => 'string|required',
+        ]);
+
+        if (!config('rapidez.vat-validation.allow-anywhere')) {
+            // Check if we should validate here
+            abort_unless($request->bearerToken() || $request->headers->get('referer') === route('account.register'), 401);
+
+            if ($request->bearerToken()) {
+                config('rapidez.models.quote')::whereQuoteIdOrCustomerToken($request->bearerToken())->firstOrFail();
+            }
+        }
+
+        try {
+            // Try validating the number
+            $validator = new Validator;
+            $result = Cache::remember('vat_' . $request->id, config('rapidez.vat-validation.cache-duration'), function () use ($request, $validator) {
+                return $validator->validateVatNumber($request->id);
+            });
+
+            return ['result' => $result];
+        } catch (ViesException $exception) {
+            return ['error' => $exception->getMessage()];
```

Comment:
Maybe just throw a http status code?

Reply:
Not sure which http status code would fit here. Usually this will error out because the VIES service is unavailable (i.e. `MS_MAX_CONCURRENT_REQ`). Could do a 503 but I feel like that's not really meant for this.

Reply:
anything in the 5xx range would probably be good.
We should probably return proper response codes indeed because we need to handle these properly on the client side as well.

From what i can see we currently have:
$request->validate returns 4xx with a message attached
bearer/referrer check returns 4xx without a message
whereQuoteIdOrCustomerToken returns 4xx (not certain wether with a message or not)
Validator returns 2xx even if validation fails or gets an exception, we should probably pick a relevant response code for that: https://it-tools.tech/http-status-codes probably something in the 4xx range for incorrect VAT id and 5xx for server errors.